Add a regression test that the same path is used for different data types

szewai opened the pull request:
#### 5c95919fc202a5bac158805082b5cf3ef47fed5f
<pre>
Add a regression test that the same path is used for different data types
<a href="https://bugs.webkit.org/show_bug.cgi?id=242149">https://bugs.webkit.org/show_bug.cgi?id=242149</a>

Reviewed by Chris Dumez.

This is a legit use case (some of our clients do this) that is not covered by existing tests.

* Tools/TestWebKitAPI/Tests/WebKitCocoa/WebsiteDataStoreCustomPaths.mm:
(TEST):

Canonical link: <a href="https://commits.webkit.org/252090@main">https://commits.webkit.org/252090@main</a>
</pre>

cdumez commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/WebsiteDataStoreCustomPaths.mm:
```diff
@@ -1082,3 +1082,48 @@ HTTPServer server([&] (Connection connection) {
     TestWebKitAPI::Util::run(&done);
 }
+
+TEST(WKWebsiteDataStoreConfiguration, SameCustomPathForDifferentTypes)
```
Seems like your new test is failing on iOS.

szewai replied:
Yes, this test result (based on 251969@main) is expected before the fix (251983@main)

cdumez reviewed (approved):
OK

szewai committed:
Add a regression test that the same path is used for different data types
https://bugs.webkit.org/show_bug.cgi?id=242149

Reviewed by Chris Dumez.

This is a legit use case (some of our clients do this) that is not covered by existing tests.

* Tools/TestWebKitAPI/Tests/WebKitCocoa/WebsiteDataStoreCustomPaths.mm:
(TEST):

Canonical link: https://commits.webkit.org/252090@main
